File: backend/app/schemas/site_content.py
from datetime import datetime
from pydantic import BaseModel


class SiteContentBase(BaseModel):
    hero_media_id: int | None = None
    featured_sermon_id: int | None = None
    featured_event_id: int | None = None
    about_media_id: int | None = None
    ministries_media_id: int | None = None
    contact_media_id: int | None = None
    social_facebook_url: str | None = None
    social_instagram_url: str | None = None
    social_youtube_url: str | None = None
    social_tiktok_url: str | None = None


class SiteContentUpdate(BaseModel):
    hero_media_id: int | None = None
    featured_sermon_id: int | None = None
    featured_event_id: int | None = None
    about_media_id: int | None = None
    ministries_media_id: int | None = None
    contact_media_id: int | None = None
    social_facebook_url: str | None = None
    social_instagram_url: str | None = None
    social_youtube_url: str | None = None
    social_tiktok_url: str | None = None


class SiteContentOut(SiteContentBase):
    id: int
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/services/site-content.ts
import { apiFetch } from "@/services/api";

export type SiteContent = {
  id: number;
  hero_media_id?: number | null;
  featured_sermon_id?: number | null;
  featured_event_id?: number | null;
  about_media_id?: number | null;
  ministries_media_id?: number | null;
  contact_media_id?: number | null;
  social_facebook_url?: string | null;
  social_instagram_url?: string | null;
  social_youtube_url?: string | null;
  social_tiktok_url?: string | null;
};

export async function fetchSiteContent() {
  return apiFetch("/site-content");
}

export async function updateSiteContent(payload: Partial<SiteContent>, token: string) {
  return apiFetch("/site-content", {
    method: "PUT",
    body: JSON.stringify(payload),
    token
  });
}
